src/eos-updater-data.c

```diff
@@ -22,17 +22,31 @@
 
 #include "eos-updater-data.h"
 
+#include "eos-util.h"
+
 #include <string.h>
 
-void
+gboolean
 eos_updater_data_init (EosUpdaterData *data,
-                       OstreeRepo *repo)
+                       OstreeRepo *repo,
+                       GError **error)
 {
-  g_return_if_fail (data == NULL);
-  g_return_if_fail (OSTREE_IS_REPO (repo));
+  if (data == NULL)
+    return TRUE;
+
+  g_return_val_if_fail (OSTREE_IS_REPO (repo), FALSE);
 
   memset (data, 0, sizeof *data);
   data->repo = g_object_ref (repo);
+  data->download_order = g_new (EosUpdaterDownloadSource, 1);
+  data->download_order[0] = EOS_UPDATER_DOWNLOAD_MAIN;
```

Comment:
Any reason to store this as a pointer? I didn't see it checked for `NULL` anywhere. In addition `n_download_sources` is only ever 1 or 0.